aries-site/src/examples/templates/table-customization/components/TableCustomizationExample.js

```diff
@@ -0,0 +1,250 @@
+    hoursAvailable: 5,
+    image:
+      'https://images.unsplash.com/photo-1501594907352-04cda38ebc29?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2389&q=80',
+    role: 'Designer',
+    name: 'Greg Furuiye',
+    status: 'Online',
+  },
+  {
+    location: 'San Jose, CA',
+    hoursAvailable: 25,
+    image:
+      'https://images.unsplash.com/photo-1534430480872-3498386e7856?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
+    role: 'Designer',
+    name: 'Vicky Avalos',
+    status: 'Offline',
+  },
+  {
+    location: 'Fort Collins, CO',
+    hoursAvailable: 12,
+    image:
+      'https://images.unsplash.com/photo-1584450150050-4b9bdbd51f68?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
+    role: 'Engineer',
+    name: 'Shimi Yacobi',
+    status: 'Online',
+  },
+];
+
+const StyledTextInput = styled(TextInput).attrs(() => ({
+  'aria-labelledby': 'search-icon',
+}))``;
+
+export const TableCustomizationExample = () => {
+  const [data, setData] = useState(allData);
+  const [filtering, setFiltering] = useState(false);
+  const [filters, setFilters] = useState({});
+  const [searchFocused, setSearchFocused] = useState(false);
+  const [search, setSearch] = useState('');
+  const [columns, setColumns] = useState(COLUMNS);
+  const [visible, setVisible] = useState(() =>
+    COLUMNS.map(col => col.property),
+  );
+  const inputRef = useRef();
+  const size = useContext(ResponsiveContext);
+  useEffect(() => {
+    if (searchFocused && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [searchFocused, setSearchFocused]);
+
+  const filterData = (array, criteria, searchValue = search) => {
+    if (Object.keys(criteria).length) setFiltering(true);
+    else setFiltering(false);
+    setFilters(criteria);
+
+    let filterResults;
+    const filterKeys = Object.keys(criteria);
+    filterResults = array.filter(item =>
+      // validates all filter criteria
+      filterKeys.every(key => {
+        // ignores non-function predicates
+        if (typeof criteria[key] !== 'function') return true;
+        return criteria[key](item[key]);
+      }),
+    );
+
+    if (searchValue) {
+      filterResults = filterResults.filter(o =>
+        Object.keys(o).some(
+          k =>
+            typeof o[k] === 'string' &&
+            o[k].toLowerCase().includes(searchValue.toLowerCase()),
+        ),
+      );
+    }
+    return filterResults;
+  };
+
+  return (
+    <Box
+      fill
+      gap="medium"
+      margin="auto"
+      pad={{ horizontal: 'medium' }}
+      width={{ max: 'xxlarge' }}
+    >
+      <Header pad={{ top: 'medium' }}>
+        <Box gap="xsmall" fill="horizontal">
+          <Heading level={2} margin={{ bottom: 'small', top: 'none' }}>
+            Users
+          </Heading>
+          <Box direction="row" justify="between">
+            <Box>
+              <Box align="center" direction="row" gap="small">
```

Comment:
Trying to decide whether including the filters and actions menu is additive or distracting from the "ordering of columns."

In some aspects, I think it gets in the way of showcasing the behavior of the ordering of columns because its not immediately clear where my focus should be.

On the other hand, it is displaying the ordering of columns in the context of an "advanced table" with a handful of controls.

I think this gets back to my question about what are we desiring to show in this page? Table customization? Implementing advanced tables? Customizing Order of Table Columns? ...

Thinking aloud here...

Reply:
I thought about this a little and left it in since if it's a template that they'll use they want both, although I'm good either way. If we keep filtering/searching in this example, should it be refactored a bit to match your latest filtering example?

Reply:
I haven't come to a conclusion either way yet... still pondering.

It would be great if you could swap the filtering out with the FilteringControls. I'd like to hear how your experience is implementing it.

Reply:
I'm inclined to keep them all together, as I think any scenario wanting to customize table columns will likely want filtering. So, having an example that's closer to what we think most people will do would be better.